### DISCUSSION TITLE
Prevent auto switching the workspace

### DISCUSSION THREAD

**Discussion by gaytomycode** [nikitabobko/AeroSpace#784] [questions-and-answers]
When i switch to an app that currently has no windows in my current workspace yet it has windows in another workspace, it takes me to this workspace. So for example if i have Safari open in workspace 1 and Terminal open in workspace 2 and in workspace 2 i switch to the safari app so that i can open a safari window in workspace 2 it shifts me to workspace 1 and i have to open my new window there and then move it to worskpace 2. Any way to prevent autoswitch to a workspace with a window for the app i switched to?

**Comment by Zacke0** [comment]
This isn't a solution to your problem, just what I came up with to make this issue less frustrating. 

Essentially I made the step after opening a new window in the 'wrong' workspace a bit easier. Using your example: once I've opened a Safari window in Workspace 1, instead of having to remember what workspace I was previously in and then moving the new window to that workspace, I now trigger the workspace-back-and-forth command to switch back to Workspace 2, and then trigger this little script I wrote, which grabs the previously focused window and brings it to the currently focused workspace (Workspace 2).   

I named the script grab-previous.sh, and put it in .config/aerospace:

```
#!/bin/bash

# Get current workspace
CURRENT_WORKSPACE=$(aerospace list-workspaces --focused)

# Store it temporarily
echo $CURRENT_WORKSPACE > /tmp/aerospace_current_workspace

# Focus previous window
aerospace focus-back-and-forth

# Move the now-focused window to the stored workspace
aerospace move-node-to-workspace $CURRENT_WORKSPACE

aerospace workspace $CURRENT_WORKSPACE
```

I then call this script with hyper+enter, in a command in my main config:

cmd-ctrl-shift-alt-enter = ['exec-and-forget ~/.config/aerospace/grab-previous.sh', 'mode main']

It's not very elegant (causes a bit of flickering), but for me it pretty much resolved the issue you described.

**Comment by jasoncorso** [comment]
On a per application basis, this may also be useful.  https://nikitabobko.github.io/AeroSpace/goodies#open-a-new-window-with-applescript
